gcloud-java-pubsub/src/main/java/com/google/gcloud/pubsub/spi/PublisherApi.java

```diff
@@ -0,0 +1,498 @@
+ * distribution.
+ *
+ *    * Neither the name of Google Inc. nor the names of its
+ * contributors may be used to endorse or promote products derived from
+ * this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+/*
+ * EDITING INSTRUCTIONS
+ * This file was generated from the file google/pubsub/v1/pubsub.proto,
+ * and updates to that file get reflected here through a regular refresh process.
+ * However, manual additions are allowed because the refresh process performs
+ * a 3-way merge in order to preserve those manual additions. In order to not
+ * break the refresh process, only certain types of modifications are
+ * allowed.
+ *
+ * Allowed modifications - currently there is only one type allowed:
+ * 1. New methods (these should be added to the end of the class)
+ *
+ * Happy editing!
+ */
+package com.google.gcloud.pubsub.spi;
+
+import com.google.protobuf.Empty;
+import com.google.pubsub.v1.DeleteTopicRequest;
+import com.google.pubsub.v1.GetTopicRequest;
+import com.google.pubsub.v1.ListTopicSubscriptionsRequest;
+import com.google.pubsub.v1.ListTopicSubscriptionsResponse;
+import com.google.pubsub.v1.ListTopicsRequest;
+import com.google.pubsub.v1.ListTopicsResponse;
+import com.google.pubsub.v1.PublishRequest;
+import com.google.pubsub.v1.PublishResponse;
+import com.google.pubsub.v1.PublisherGrpc;
+import com.google.pubsub.v1.PubsubMessage;
+import com.google.pubsub.v1.Topic;
+
+import io.gapi.gax.grpc.ApiCallable;
+import io.gapi.gax.grpc.PageDescriptor;
+import io.gapi.gax.grpc.ServiceApiSettings;
+import io.gapi.gax.internal.ApiUtils;
+import io.gapi.gax.protobuf.PathTemplate;
+import io.grpc.Channel;
+import io.grpc.ManagedChannel;
+
+import java.io.IOException;
+import java.util.List;
+
+// Manually-added imports: add custom (non-generated) imports after this point.
+
+
+
+// AUTO-GENERATED DOCUMENTATION AND SERVICE - see instructions at the top of the file for editing.
+/**
+ * The service that an application uses to manipulate topics, and to send
+ * messages to a topic.
+ */
+@javax.annotation.Generated("by the veneer generator")
+public class PublisherApi implements AutoCloseable {
```

Comment:
Mocking frameworks these days allow you to override a lot, even when you wouldn't be able to in normal code. As for my statement about all methods being overridable, you are correct - I was being imprecise; I meant that they are by default. I will make a modification so that the access levels will allow the overriding that is necessary.